Content Tree: Keyboard navigation to move items

ElishaSamPeterPrabhu opened the pull request:
## Description

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

- Added Keyboard navigation to move items from one position to another

References 
Fixes #1597 <!-- issue number -->

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

<!--Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration -->

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

cjwinsor commented:
@ElishaSamPeterPrabhu 2 things:
When i pick an item and press down once, it does not move to where i expect it to go, it essentially is putting the line where the item already is, requiring that i press down twice.  Can we have it so the down arrow is aware enough to put it where we expect?

Can we handle escape key to cancel as well?

ElishaSamPeterPrabhu commented:
> @ElishaSamPeterPrabhu 2 things: When i pick an item and press down once, it does not move to where i expect it to go, it essentially is putting the line where the item already is, requiring that i press down twice. Can we have it so the down arrow is aware enough to put it where we expect?
> 
> Can we handle escape key to cancel as well?

Updated the up and down behavior as expected and added escape to cancel draggable state
